### PR TITLE
add `milli/micro-seconds` to RelTime and fix doc

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**
CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: docs/source/daml/reference/data-types.rst [hunk 1]
```diff
@@ -53,15 +53,16 @@ Table of built-in primitive types
    * - ``Date``
      - models dates
      - ``date 2007 Apr 5``
-     - To create a value of type ``Date``, use the function ``date`` (to get this function, import ``DA.Date``).
+     - Permissible dates range from ``0001-01-01`` to ``9999-12-31`` (using a year-month-day format). To create a value of type ``Date``, use the function ``date`` (to get this function, import ``DA.Date``).
    * - ``Time``
      - models absolute time (UTC)
      - ``time (date 2007 Apr 5) 14 30 05``
-     - ``Time`` values have microsecond precision. To create a value of type ``Time``, use a ``Date`` and the function ``time`` (to get this function, import ``DA.Time``).
+     - ``Time`` values have microsecond precision with allowed range from ``0001-01-01`` to ``9999-12-31`` and using a year-month-day format. To create a value of type ``Time``, use a ``Date`` and the function ``time`` (to get this function, import ``DA.Time``).
```
Comment (remyhaemmerle-da) [inline docs/source/daml/reference/data-types.rst:60]:
```suggestion
     - ``Time`` values have microsecond precision with allowed range from ``0001-01-01`` to ``9999-12-31`` (using a year-month-day format). To create a value of type ``Time``, use a ``Date`` and the function ``time`` (to get this function, import ``DA.Time``).
```

File: compiler/damlc/daml-stdlib-src/DA/Time.daml [hunk 2]
```diff
@@ -58,6 +60,14 @@ minutes x = seconds (60 * x)
 seconds : Int -> RelTime
 seconds x = convertMicrosecondsToRelTime (1000000 * x)
 
+-- A number of milliseconds in relative time.
```
Comment (sofiafaro-da) [inline compiler/damlc/daml-stdlib-src/DA/Time.daml:63]: So it shows up in the docs:

```suggestion
-- | A number of milliseconds in relative time.
```

File: compiler/damlc/daml-stdlib-src/DA/Time.daml [hunk 3]
```diff
@@ -58,6 +60,14 @@ minutes x = seconds (60 * x)
...
+milliseconds: Int -> RealTime
+milliseconds x = convertMicrosecondsToRelTime (1000 * x)
+
+-- A number of microseconds in relative time.
```
Comment (sofiafaro-da) [inline compiler/damlc/daml-stdlib-src/DA/Time.daml:67]:
```suggestion
-- | A number of microseconds in relative time.
```

File: compiler/damlc/daml-stdlib-src/DA/Time.daml (hunk 2 above)
Comment (sofiafaro-da) [reply compiler/damlc/daml-stdlib-src/DA/Time.daml:63]: Worth changing for `seconds` too.

**Review by sofiafaro-da — APPROVED** [review]
LGTM, thank you.

File: compiler/damlc/daml-stdlib-src/DA/Time.daml (hunk 2 above)
Comment (cocreature) [inline compiler/damlc/daml-stdlib-src/DA/Time.daml:63]: can we add some tests for this? maybe seems simple enough but I’d still prefer to err on the safe side.

**Review by cocreature — APPROVED** [review]
Nice thank you!

File: compiler/damlc/daml-stdlib-src/DA/Time.daml (hunk 2 above)
Comment (remyhaemmerle-da) [reply compiler/damlc/daml-stdlib-src/DA/Time.daml]: :+1:

File: compiler/damlc/daml-stdlib-src/DA/Time.daml [hunk 4]
```diff
@@ -54,10 +56,18 @@ hours x = minutes (60 * x)
 minutes : Int -> RelTime
 minutes x = seconds (60 * x)
 
--- A number of seconds in relative time.
+-- | A number of seconds in relative time.
 seconds : Int -> RelTime
 seconds x = convertMicrosecondsToRelTime (1000000 * x)
 
+-- | A number of milliseconds in relative time.
+milliseconds: Int -> RelTime
```
Comment (sofiafaro-da) [inline compiler/damlc/daml-stdlib-src/DA/Time.daml:64]: nitpick: daml style is closer to Haskell :) 

```suggestion
milliseconds : Int -> RelTime
```

File: compiler/damlc/daml-stdlib-src/DA/Time.daml [hunk 5]
```diff
@@ -54,10 +56,18 @@ hours x = minutes (60 * x)
...
+milliseconds x = convertMicrosecondsToRelTime (1000 * x)
+
+-- | A number of microseconds in relative time.
+microseconds: Int -> RelTime
```
Comment (sofiafaro-da) [inline compiler/damlc/daml-stdlib-src/DA/Time.daml:68]:
```suggestion
microseconds : Int -> RelTime
```

File: compiler/damlc/daml-stdlib-src/DA/Time.daml [hunk 6]
```diff
@@ -53,11 +53,11 @@ hours : Int -> RelTime
 hours x = minutes (60 * x)
 
 -- | A number of minutes in relative time.
-minutes : Int -> RelTime
+minutes: Int -> RelTime
 minutes x = seconds (60 * x)
 
 -- | A number of seconds in relative time.
-seconds : Int -> RelTime
```
Comment (sofiafaro-da) [inline compiler/damlc/daml-stdlib-src/DA/Time.daml:58]: Why are you moving further away from the style in the rest of the standard library? Please revert.

Comment (remyhaemmerle-da) [reply compiler/damlc/daml-stdlib-src/DA/Time.daml:58]: My bad. :)

**Pull request merged**